### PR TITLE
Figure.plot: Deprecate parameter "sizes" to "size" (remove in v0.6.0)

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

- Rename the parameter `sizes` to `size` in Figure.plot()
- Updates tests and examples to use the new parameter name
- Use the deprecate_parameter decorator for backward-compatibility
- Add a test for checking 'sizes' backward compatibility

Address #1118.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/tutorials/plot.py [hunk 1]
```diff
@@ -82,7 +82,7 @@
 fig.plot(
     x=data.longitude,
     y=data.latitude,
-    sizes=0.02 * 2 ** data.magnitude,
+    size=0.02 * 2 ** data.magnitude,
```
Comment (core-man) [inline examples/tutorials/plot.py:85]: Line 66:
> So in this case, the sizes

`sizes` to be `size`?

**Review by core-man — APPROVED** [review]
All are good except one minor comment.

**Review by michaelgrund — APPROVED** [review]
Except the comment provided by @core-man, everything looks fine to me!

File: pygmt/src/plot.py [hunk 2]
```diff
@@ -78,7 +80,7 @@ def plot(self, x=None, y=None, data=None, sizes=None, direction=None, **kwargs):
         Either a data file name or a 2d numpy array with the tabular data.
         Use parameter ``columns`` to choose which columns are x, y, color,
         and size, respectively.
-    sizes : 1d array
+    size : 1d array
         The sizes of the data points in units specified using ``style``.
```
Comment (weiji14) [inline pygmt/src/plot.py:84]:
```suggestion
        The size of the data points in units specified using ``style``.
```

File: pygmt/src/plot.py [hunk 3]
```diff
@@ -215,12 +217,12 @@ def plot(self, x=None, y=None, data=None, sizes=None, direction=None, **kwargs):
             )
         extra_arrays.append(kwargs["G"])
         del kwargs["G"]
-    if sizes is not None:
+    if size is not None:
         if kind != "vectors":
             raise GMTInvalidInput(
                 "Can't use arrays for sizes if data is matrix or file."
```
Comment (weiji14) [inline pygmt/src/plot.py:223]:
```suggestion
                "Can't use arrays for 'size' if data is a matrix or file."
```

File: pygmt/tests/test_plot.py [hunk 4]
```diff
@@ -152,7 +153,7 @@ def test_plot_sizes(data, region):
     fig.plot(
```
Comment (weiji14) [inline pygmt/tests/test_plot.py:153]: Can you also rename other instances of 'sizes' to 'size' in the `""" """` docstring? E.g. "Plot the data using z as sizes." becomes "Plot the data using z as size.". Also debating whether to keep the test names as `test_plot_*_sizes` or rename to `test_plot_*_size` (which would mean renaming a lot of *.png.dvc files).

File: examples/tutorials/plot.py [hunk 5]
```diff
@@ -54,7 +54,7 @@
 fig.plot(
     x=data.longitude,
     y=data.latitude,
-    sizes=0.02 * (2 ** data.magnitude),
+    size=0.02 * (2 ** data.magnitude),
     style="cc",
     color="white",
     pen="black",
```
Comment (weiji14) [inline examples/tutorials/plot.py:60]: Please also change 'sizes' to 'size' at Line 66 in this `examples/tutorials/plot.py` file.

File: pygmt/tests/test_plot.py (hunk 4 above)
Comment (seisman) [reply pygmt/tests/test_plot.py:153]: > E.g. "Plot the data using z as sizes." becomes "Plot the data using z as size."

In this sentence, "sizes" means "sizes" of symbols, not the `size` parameter. I think it is grammatically correct, right? 

BTW, we also use "Plot the data using z as colors" even though the parameter name is `color`.

File: examples/tutorials/plot.py (hunk 5 above)
Comment (seisman) [reply examples/tutorials/plot.py:60]: Similarly, at line 66, the sentence is about the symbol sizes, not the `size` parameter.

Comment (weiji14) [reply examples/tutorials/plot.py:60]: The full paragraph from L64-66 reads:

```
# Notice that we didn't include the size in the ``style`` parameter this time, just the
# symbol ``c`` (circles) and the unit ``c`` (centimeter). So in this case, the sizes
# will be interpreted as being in centimeters.
```

So either change the `size` in L64 to `sizes`, or the `sizes` in L66 to `size`?

File: pygmt/tests/test_plot.py (hunk 4 above)
Comment (weiji14) [reply pygmt/tests/test_plot.py:153]: Hmm, I think either `sizes` or `size` is grammatically correct. Let's just keep it on `sizes` to reduce the `git diff` then.

**Comment by seisman** [comment]
Since the PR title will be used in the release notes, I just changed the PR title to 

> Figure.plot: Deprecate parameter "sizes" to "size" (will be removed in v0.6.0)

to make it more readable. 

I expect to have a "consistent" PR title for parameter deprecations like this, so suggestions on the PR title format are also welcomed.

**Review by weiji14 — APPROVED** [review]
> Since the PR title will be used in the release notes, I just changed the PR title to
> 
> > Figure.plot: Deprecate parameter "sizes" to "size" (will be removed in v0.6.0)
> 
> to make it more readable.
> 
> I expect to have a "consistent" PR title for parameter deprecations like this, so suggestions on the PR title format are also welcomed.

Looks good. I would just change the 'removed' to 'remove' in the title, i.e. 'Figure.plot: Deprecate parameter "sizes" to "size" (remove in v0.6.0)'

File: pygmt/tests/test_plot.py [hunk 6]
```diff
@@ -446,3 +447,28 @@ def test_plot_datetime():
...
+            y=data[:, 1],
+            sizes=0.5 * data[:, 2],
+            region=region,
+            projection="X10c",
+            style="cc",
+            color="blue",
+            frame="af",
+        )
+        assert len(w) == 1
+        assert issubclass(w[0].category, FutureWarning)
```
Comment (weiji14) [inline pygmt/tests/test_plot.py:473]: Actually, would it be better to use `pytest.warns` here? See https://docs.pytest.org/en/6.2.x/warnings.html#warns.

Comment (weiji14) [reply pygmt/tests/test_plot.py:473]: Might need to remove the `import warnings` line at the top.

```suggestion
    with pytest.warns(expected_warning=FutureWarning) as record:
        fig.plot(
            x=data[:, 0],
            y=data[:, 1],
            sizes=0.5 * data[:, 2],
            region=region,
            projection="X10c",
            style="cc",
            color="blue",
            frame="af",
        )
        assert len(record) == 1  # check that only one warning was raised
```

File: pygmt/tests/test_plot.py [hunk 7]
```diff
@@ -4,6 +4,7 @@
 """
 import datetime
 import os
+import warnings
```
Comment (seisman) [inline pygmt/tests/test_plot.py:7]:
```suggestion
import os
```

Comment (seisman) [inline pygmt/tests/test_plot.py:7]:
```suggestion
```

**Pull request merged**